src/clone.d

```diff
@@ -811,37 +834,57 @@ extern (C++) FuncDeclaration buildPostBlit(StructDeclaration sd, Scope* sc)
             continue;
         sdv.dtor.functionSemantic();
 
-        ex = new ThisExp(loc);
-        ex = new DotVarExp(loc, ex, v);
-        if (v.type.toBasetype().ty == Tstruct)
+        tv = v.type.toBasetype();
+        if (tv.ty == Tstruct)
         {
             // this.v.__xdtor()
+
+            ex = new ThisExp(loc);
+            ex = new DotVarExp(loc, ex, v);
+
             // This is a hack so we can call destructors on const/immutable objects.
             ex = new AddrExp(loc, ex);
             ex = new CastExp(loc, ex, v.type.mutableOf().pointerTo());
             ex = new PtrExp(loc, ex);
             if (stc & STCsafe)
                 stc = (stc & ~STCsafe) | STCtrusted;
+
             ex = new DotVarExp(loc, ex, sdv.dtor, false);
             ex = new CallExp(loc, ex);
         }
         else
         {
             // _ArrayDtor((cast(S*)this.v.ptr)[0 .. n])
+
+            uinteger_t n = 1;
+            while (tv.ty == Tsarray)
+            {
+                n *= (cast(TypeSArray)tv).dim.toUInteger();
+                tv = tv.nextOf().toBasetype();
+            }
+            //if (n == 0)
+            //    continue;
```

Comment:
No, the `n == 0` check in previous `_ArrayPostblit` block will completely skip this code section. Therefore empty array never be destructed..